timeseries/src/autogluon/timeseries/dataset/ts_dataframe.py

```diff
@@ -921,7 +921,11 @@ def train_test_split(
         test_data : TimeSeriesDataFrame
             Test portion of the data. Contains the slice ``[:end_idx]`` of each time series in the original dataset.
         """
-        test_data = self.slice_by_timestep(None, end_index)
+        df = self
+        if not df.index.is_monotonic_increasing:
+            logger.warning("Sorting the dataframe index before generating the train/test split.")
+            df = df.sort_index()
```

Comment:
does this also sort by item_ids? or do we assume it was already sorted?

Reply:
Yes, we also sort by item_ids because there is no efficient way to check that all timestamps are increasing other than `index.is_monotonic_increasing` (which also checks that item_ids are sorted).

Inside the TimeSeriesPredictor, we anyway always sort by item_id + timestamps, so this change will only affect users that are using `tsdf.train_test_split` outside of the predictor.